dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -150,30 +227,264 @@ def _common_type(*arrays):
+
+    Returns
+    -------
+    tuple:
+        lu_t : (..., N, N) {dpnp.ndarray, usm_ndarray}
+            Combined 'L' and 'U' matrices from LU decomposition
+            excluding the diagonal of 'L'.
+        piv : (..., N) {dpnp.ndarray, usm_ndarray}
+            1-origin pivot indices indicating row permutations during decomposition.
+        dev_info : (...) {dpnp.ndarray, usm_ndarray}
+            Information on `getrf` or `getrf_batch` computation success (0 for success).
+
+    """
+
+    n = a.shape[-2]
+
+    a_sycl_queue = a.sycl_queue
+    a_usm_type = a.usm_type
+
+    # TODO: Find out at which array sizes the best performance is obtained
+    # getrf_batch implementation shows slow results with large arrays on GPU.
+    # Use getrf_batch only on CPU.
+    # On GPU call getrf for each two-dimensional array by loop
+    use_batch = a.sycl_device.has_aspect_cpu
+
+    if a.ndim > 2:
+        orig_shape = a.shape
+        # get 3d input arrays by reshape
+        a = a.reshape(-1, n, n)
+        batch_size = a.shape[0]
+        a_usm_arr = dpnp.get_usm_ndarray(a)
+
+        if use_batch:
+            # `a` must be copied because getrf_batch destroys the input matrix
+            a_h = dpnp.empty_like(a, order="C", dtype=res_type)
+            ipiv_h = dpnp.empty(
+                (batch_size, n),
+                dtype=dpnp.int64,
+                usm_type=a_usm_type,
+                sycl_queue=a_sycl_queue,
+            )
+            dev_info_h = [0] * batch_size
+
+            a_ht_copy_ev, a_copy_ev = ti._copy_usm_ndarray_into_usm_ndarray(
+                src=a_usm_arr, dst=a_h.get_array(), sycl_queue=a_sycl_queue
+            )
+
+            ipiv_stride = n
+            a_stride = a_h.strides[0]
+
+            # Call the LAPACK extension function _getrf_batch
+            # to perform LU decomposition of a batch of general matrices
+            ht_lapack_ev, _ = li._getrf_batch(
+                a_sycl_queue,
+                a_h.get_array(),
+                ipiv_h.get_array(),
+                dev_info_h,
+                n,
+                a_stride,
+                ipiv_stride,
+                batch_size,
+                [a_copy_ev],
+            )
+
+            ht_lapack_ev.wait()
+            a_ht_copy_ev.wait()
+
+            dev_info_array = dpnp.array(
+                dev_info_h, usm_type=a_usm_type, sycl_queue=a_sycl_queue
+            )
+
+            # Reshape the results back to their original shape
+            a_h = a_h.reshape(orig_shape)
+            ipiv_h = ipiv_h.reshape(orig_shape[:-1])
+            dev_info_array = dev_info_array.reshape(orig_shape[:-2])
+
+            return (a_h, ipiv_h, dev_info_array)
+
+        else:
+            # Initialize lists for storing arrays and events for each batch
+            a_vecs = [None] * batch_size
+            ipiv_vecs = [None] * batch_size
+            dev_info_vecs = [None] * batch_size
```

Comment:
This will not work because python will create a list of (`batch_size`) elements where each element is a reference to the same nested list `[0]` and changing it will change the whole list
```
a = [[0]] * 3
a[0][0] = 1
a
Out[33]: [[[1]], [[1]], [[1]]]

```